src/OMSimulatorLib/ComponentTable.cpp

```diff
@@ -216,6 +216,56 @@ oms_status_enu_t oms::ComponentTable::getReal(const oms::ComRef& cref, double& v
   return logError("out of range (cref=" + std::string(cref) + ", time=" + std::to_string(time) + ")");
 }
 
+oms_status_enu_t oms::ComponentTable::getInteger(const oms::ComRef& cref, int& value)
+{
+  if (!resultReader)
+    logError("the table isn't initialized properly");
+
+  if (series.find(cref) == series.end())
+    series[cref] = resultReader->getSeries(cref.c_str());
+
+  for (int i=1; i<series[cref]->length; ++i)
+  {
+    if (series[cref]->time[i-1] == time)
+    {
+      value = series[cref]->value[i-1];
+      return oms_status_ok;
+    }
+    else if (series[cref]->time[i-1] <= time && series[cref]->time[i] >= time)
+    {
+      value = series[cref]->value[i-1];
+      return oms_status_ok;
+    }
+  }
+  value = 0;
+  return logError("out of range (cref=" + std::string(cref) + ", time=" + std::to_string(time) + ")");
+}
+
+oms_status_enu_t oms::ComponentTable::getBoolean(const oms::ComRef& cref, bool& value)
+{
+  if (!resultReader)
+    logError("the table isn't initialized properly");
+
+  if (series.find(cref) == series.end())
+    series[cref] = resultReader->getSeries(cref.c_str());
+
+  for (int i=1; i<series[cref]->length; ++i)
+  {
+    if (series[cref]->time[i-1] == time)
+    {
+      value = series[cref]->value[i-1];
+      return oms_status_ok;
+    }
+    else if (series[cref]->time[i-1] <= time && series[cref]->time[i] >= time)
+    {
+      value = series[cref]->value[i-1];
+      return oms_status_ok;
+    }
+  }
```

Comment:
I think this logic is wrong for the last point of a table. It only works if using interpolation as we do for the implementation of continuous signals.

Reply:
@lochel Yes you are correct , the last point of a table will not work

Reply:
Also the return value is not different for the two if cases. So one could replace it by one if-statement.
```c
for (int i=1; i<series[cref]->length; ++i)
{
    if (series[cref]->time[i-1] <= time && series[cref]->time[i] >= time)
    {
        value = series[cref]->value[i-1];
        return oms_status_ok;
    }
}
```

Reply:
@lochel  this problem should be fixed by this https://github.com/OpenModelica/OMSimulator/pull/785/commits/07ca3bfb305a88ed9ee8878525df6c0d5989da8f